style/hudl_sql_style_guide.md

```diff
@@ -5,33 +5,101 @@ description: A guide to writing clean, clear, and consistent SQL.
-* Always use AS when creating aliases for columns, tables, subqueries
-* Always capitalize SQL keywords (e.g., `SELECT` or `AS`)
-* Always capitalize SQL datatypes (e.g., `VARCHAR` or `DATE`)
-* Comments should go near the top of your query, or at least near the closest `SELECT`
-* Try to only comment on things that aren't obvious about the query (e.g., why a particular ID is hardcoded, etc.)
-* All dates should be formatted as 'YYYY-MM-DD'
-* Use `!=` for the inequality operator
-
-## Naming Conventions
-
+* We believe consistency in style is very important
+* We demonstrate intent explicitly in code, via clear structure and comments where needed
+* We value DRY (do not repeat yourself) code and readability, but will favor readability over DRY when the two are in conflict
+* We believe everyone is responsible for code and modeling quality
+
+## Data Modeling
+### Data Modeling: The process of using business logic to aggregate or otherwise transform raw data
+### Data Model: A structure that organizes and standardizes data elements and how they relate to each other and a real-world entity or concept. In our case, a data model is a standardized representation of a business entity or concept. 
+
+When it has been determined that a new data model is required to be built, the person building the new model will be required to hold a data model design review. This is a process in which the builder of the data model will write-up their design plans and hold a meeting for these plans to be peer reviewed. 
+
+The purpose of this is not to block or slow work. Rather, this is a means to ensure data quality, prevent painful re-work and identify dependencies that may have been overlooked. Each data model design will require the following:
+
+* __State the purpose__: 
+    Every new data model that is introduced must have a clear purpose linked to business value. All models added should be done so with a high amount of rigor to       ensure data quality.
+* __Start With the End in Mind__: 
+    Test Driven Development https://www.guru99.com/test-driven-development.html is when the tests are written before the code. While we don’t require a full set of     all tests in the Data Model Design process, an overview of test cases that will be considered and implemented should be included as a part of the data model       design review.
+* __Identify the Grains__: 
+    Identify what the grains will be of the final table and of any tables from which you will be creating your final table.
+* Sample of table output with examples (can be hypothetical)
+* Column names and descriptions defined 
+
+### File Orgainzation
+* __Business-centric__ models and __source-centric__ models should be organizationally distinct. __Business-centric__ models define business logic. __Source-centric__ models select from dbt sources and their primary function is to clean and group like-data (ie: payment data may come from two different sources, but need to be joined).
```

Comment:
`Source` in dbt could refer to actual sources. Based on the description, I'm not sure if that's entirely what you're describing here. The example of payments data could be more of an intermediate model.